addons/mail/static/src/js/views/activity/activity_renderer.js

```diff
@@ -173,141 +147,65 @@ var ActivityRenderer = AbstractRenderer.extend({
-            var activity_group = self.state.grouped_activities[resId];
-            activity_group = activity_group && activity_group[activity_type_id] || {count: 0, ids: [], state: false};
-
-            var $td = $(QWeb.render('mail.ActivityViewRow', {
-                resId: resId,
-                activityGroup: activity_group,
-                activityTypeId: activity_type_id,
-                widget: self,
-            }));
-            if (activity_group.state) {
-                var record = self.getKanbanActivityData(activity_group, resId);
-                var widget = new KanbanActivity(self, "activity_ids", record, {});
-                var def = widget.appendTo($td).then(function () {
-                    // replace clock by closest deadline
-                    var $date = $('<div class="o_closest_deadline">');
-                    // split the year/month/day to create a date (to have a date, no tz, instead of a datetime, tz dependant)
-                    var splitDate = activity_group.o_closest_deadline.split('-');
-                    // the value representing the month, ranges from 0 for January up to 11 for December.
-                    var date = new Date(splitDate[0], splitDate[1]-1, splitDate[2]);
-                    // To remove year only if current year
-                    if (moment().year() === moment(date).year()) {
-                        $date.text(date.toLocaleDateString(moment().locale(), { day: 'numeric', month: 'short' }));
-                    } else {
-                        $date.text(moment(date).format('ll'));
-                    }
-                    $td.find('a').html($date);
-                    if (activity_group.count > 1) {
-                        $td.find('a').append($('<span>', {
-                            class: 'badge badge-light badge-pill border-0 ' + activity_group.state,
-                            text: activity_group.count,
-                        }));
-                    }
-                });
-                defs.push(def);
-            }
-            return $td;
-        });
-        return Promise.all(defs).then(function () {
-            var $tr = $('<tr/>', {class: 'o_data_row'}).attr('data-res-id', resId)
-                .append($nameTD)
-                .append($cells);
-            return $tr;
-        });
-    },
+        }
+    }
 
     //--------------------------------------------------------------------------
     // Handlers
     //--------------------------------------------------------------------------
-
+    
     /**
      * @private
      * @param {MouseEvent} ev
      */
-    _onEmptyCell: function (ev) {
-        ev.preventDefault();
-        ev.stopPropagation();
-
-        var data = $(ev.currentTarget).data();
-        this.trigger_up('empty_cell_clicked', data);
-    },
+    _onEmptyCellClicked(ev) {
+        this.trigger('empty_cell_clicked', {
+            resId: parseInt(ev.currentTarget.dataset.resId, 10),
```

Comment:
check target currentTarget

Reply:
> target = the element that triggers the event
> currentTarget = the element that listens and has catched the event

In my case the listener is on a \<div> and it has a \<i> child so it's better to use currentTarget

Reply:
:+1: